eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentLifecycle.scala

```diff
@@ -312,12 +307,15 @@ object PaymentLifecycle {
   /**
    * Send a payment to a pre-defined route without running the path-finding algorithm.
    *
-   * @param hops         payment route to use.
+   * @param route        payment route to use.
    * @param finalPayload onion payload for the target node.
    */
-  case class SendPaymentToRoute(hops: Seq[PublicKey], finalPayload: FinalPayload, assistedRoutes: Seq[Seq[ExtraHop]] = Nil) {
-    require(hops.nonEmpty, s"payment route must not be empty")
-    val targetNodeId = hops.last
+  case class SendPaymentToRoute(route: Either[Seq[PublicKey], Route], finalPayload: FinalPayload, assistedRoutes: Seq[Seq[ExtraHop]] = Nil) {
```

Comment:
If you dislike the `Either`, I could also make `SendPaymentToRoute` a trait with two distinct child case classes, I don't know if it's better.

Reply:
It's fine with me that way.